packages/process/src/common/shell-quoting.spec.ts

```diff
@@ -0,0 +1,176 @@
+                    },
+                ],
+                [ShellQuoting.Weak]: [
+                    { input: 'abc', expected: '"abc"' },
+                    { input: 'ab c', expected: '"ab c"' },
+                    { input: 'ab"c', expected: '"ab\\"c"' },
+                    { input: 'ab\'c', expected: '"ab\'c"' },
+                    { input: 'ab^ c^', expected: '"ab^ c^"' },
+                    {
+                        input: 'setTimeout(() => { console.log(1, "2\'3"); }, 100)',
+                        expected: '"setTimeout(() => { console.log(1, \\"2\'3\\"); }, 100)"',
+                    },
+                    {
+                        input: 'console.log("%PATH%")',
+                        expected: '"console.log(\\"%PATH%\\")"',
+                    },
+                ]
+            },
+        },
+        // https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-6
+        powershell: {
+            quotingFunctions: PowershellQuotingFunctions,
+            data: {
+                [ShellQuoting.Escape]: [
+                    { input: 'abc', expected: 'abc' },
+                    { input: 'ab c', expected: 'ab` c' },
+                    { input: 'ab"c', expected: 'ab`"c' },
+                    { input: 'ab\'c', expected: 'ab`\'c' },
+                    { input: 'ab` c`', expected: 'ab``` c``' },
+                    {
+                        input: 'setTimeout(() => { console.log(1, "2\'3"); }, 100)',
+                        expected: 'setTimeout`(`(`)` =`>` `{` console.log`(1,` `"2`\'3`"`)`;` `},` 100`)',
+                    },
+                ],
+                [ShellQuoting.Strong]: [
+                    { input: 'abc', expected: '\'abc\'' },
+                    { input: 'ab c', expected: '\'ab c\'' },
+                    { input: 'ab"c', expected: '\'ab"c\'' },
+                    { input: 'ab\'c', expected: '\'ab\'\'c\'' },
+                    { input: 'ab` c`', expected: '\'ab` c`\'' },
+                    {
+                        input: 'setTimeout(() => { console.log(1, "2\'3"); }, 100)',
+                        expected: '\'setTimeout(() => { console.log(1, "2\'\'3"); }, 100)\'',
+                    },
+                ],
+                [ShellQuoting.Weak]: [
+                    { input: 'abc', expected: '"abc"' },
+                    { input: 'ab c', expected: '"ab c"' },
+                    { input: 'ab"c', expected: '"ab`"c"' },
+                    { input: 'ab\'c', expected: '"ab\'c"' },
+                    { input: 'ab` c`', expected: '"ab` c``"' },
+                    {
+                        input: 'setTimeout(() => { console.log(1, "2\'3"); }, 100)',
+                        expected: '"setTimeout(() => { console.log(1, `"2\'3`"); }, 100)"',
+                    },
+                ]
+            },
+        }
+    } as const;
+
+    // iter through each runtime (bash/cmd/powershell)
```

Comment:
[minor] these comments, and the ones below are inconsistent with the rest of the file, mainly the format of capitalized case ended in punctuation.